presto-spi/src/main/java/com/facebook/presto/spi/type/VarcharType.java

```diff
@@ -29,6 +29,11 @@
     public static final int MAX_LENGTH = Integer.MAX_VALUE;
     public static final VarcharType VARCHAR = new VarcharType(MAX_LENGTH);
 
+    public static VarcharType createUnboundedVarcharType()
```

Comment:
It's implementation detail only as long as the object is immutable - which is the case right now.
I will introduce this function into VARCHAR(x) functions as well than ;)